svm/src/account_loader.rs

```diff
@@ -95,6 +95,102 @@ pub struct FeesOnlyTransaction {
     pub fee_details: FeeDetails,
 }
 
+#[cfg_attr(feature = "dev-context-only-utils", derive(Clone))]
+pub(crate) struct AccountLoader<'a, CB: TransactionProcessingCallback> {
+    account_overrides: Option<&'a AccountOverrides>,
+    pub(crate) program_cache: ProgramCacheForTxBatch,
+    program_accounts: HashMap<Pubkey, (&'a Pubkey, u64)>,
+    callbacks: &'a CB,
+    pub(crate) feature_set: Arc<FeatureSet>,
+}
+impl<'a, CB: TransactionProcessingCallback> AccountLoader<'a, CB> {
+    pub fn new(
+        account_overrides: Option<&'a AccountOverrides>,
+        program_cache: ProgramCacheForTxBatch,
+        program_accounts: HashMap<Pubkey, (&'a Pubkey, u64)>,
+        callbacks: &'a CB,
+        feature_set: Arc<FeatureSet>,
+    ) -> AccountLoader<'a, CB> {
```

Comment:
instead of following the "pub first priv after" rule which made things hard to remember, the fields are in the order of load priority, overrides > program cache > accounts-db, with constructor args in the same order

the next pr immediately after this one adds account cache and removes overrides from the struct. when `disable_account_loader_special_case` activates we also remove the two program cache items from the struct